### PR TITLE
Improve people search responsiveness

### PR DESCRIPTION
**Pull request by grilix**
When searching for people, local results (contacts, and in the future we
could even include cached results) are shown immediately instead of
waiting for the API response.

### CODE REVIEW

File: src/actions/contactsActions.js [hunk 1]
```diff
@@ -48,22 +48,26 @@ import type { Dispatch, GetState } from 'reducers/rootReducer';
...
 export const searchContactsAction = (query: string) => {
   return async (dispatch: Dispatch, getState: GetState, api: Object) => {
-    const { user: { data: { walletId } }, contacts: { data: localContacts } } = getState();
+    const {
+      user: { data: { walletId } },
+      contacts: { data: localContacts },
+    } = getState();
+    const upperCaseQuery = query.toUpperCase();
+
+    const myContacts = localContacts.filter(({ username }) => {
```
Comment (poocart) [inline src/actions/contactsActions.js]:
```suggestion
    const myContacts = localContacts.filter(({ username }) => username.toUpperCase().includes(upperCaseQuery));
```

File: src/actions/contactsActions.js [hunk 2]
```diff
@@ -48,22 +48,26 @@ import type { Dispatch, GetState } from 'reducers/rootReducer';
...
+      return username.toUpperCase().includes(upperCaseQuery);
```
Comment (poocart) [inline src/actions/contactsActions.js]:
```suggestion
```

File: src/actions/contactsActions.js [hunk 3]
```diff
@@ -48,22 +48,26 @@ import type { Dispatch, GetState } from 'reducers/rootReducer';
...
+    });
```
Comment (poocart) [inline src/actions/contactsActions.js]: Per suggestion above:
```suggestion
```

File: src/actions/contactsActions.js (hunk 1 above)
Comment (poocart) [reply src/actions/contactsActions.js]: Or you can even use `isCaseInsensitiveMatch` from utils.

Comment (grilix) [reply src/actions/contactsActions.js]: Well actually, no, `isCaseInsensitiveMatch` doesn't provide the same behaviour :grimacing:

**Pull request merged**